pkg/webhook/validator.go

```diff
@@ -382,55 +384,56 @@ func ValidatePolicy(ctx context.Context, namespace string, ref name.Reference, c
 			switch {
 			case authority.Static != nil:
 				if authority.Static.Action == "fail" {
-					result.err = errors.New("disallowed by static policy")
+					result.err = cosign.NewVerificationError("disallowed by static policy")
 					results <- result
 					return
 				}
-				result.signatures = []PolicySignature{{Subject: "allowed by static policy", Issuer: "allowed by static policy"}}
```

Comment:
This was necessary because of the `default` case below, which errors when both `result.signatures` and `result.attestations` were empty.  I've changed that to allow this case, so that we don't have to look for these non-`const` sentinel strings.

Reply:
I see what you're doing here but having a default below that is fail open just makes me a bit nervous (not in this PR), but in general.
What do you think about changing this structure here and add a field for Static pass instead:
https://github.com/sigstore/policy-controller/blob/main/pkg/webhook/validator_result.go#L40

Say, AuthorityMatch.StaticPass

Then we could tag this as having satisfied that and check for that before the default clause and leave default be fail?

Just thinking again for future and safety?

Reply:
As long as we're not lying about it having signatures, I'm happy 😁

Reply:
Yeah, that's why think having a new field that's not lying is good. I probably should've just added it the first time around :)